pkg/virt-handler/vm.go

```diff
@@ -2290,3 +2298,23 @@ func isACPIEnabled(vmi *v1.VirtualMachineInstance, domain *api.Domain) bool {
 		domain.Spec.Features != nil &&
 		domain.Spec.Features.ACPI != nil
 }
+
+func setMissingSRIOVInterfacesNames(sriovInterfacesNameByMac map[string]string, interfaceStatusByMac map[string]api.InterfaceStatus) {
```

Comment:
Since `updateVMIStatus` is already long and complex, I would move all the sriov logic to one method (`indexSpecIfaceNamesByMac` can be removed).
```
func setMissingSRIOVInterfacesNames(interfacesSpecByName map[string]v1.Interface, interfaceStatusByMac map[string]api.InterfaceStatus) {
	for name, ifaceSpec := range interfacesSpecByName {
		if specIface.SRIOV == nil || ifaceSpec.MacAddress == "" {
			continue;
		}
		if domainIfaceStatus, exists := interfaceStatusByMac[ifaceSpec.MacAddress]; exists {
			domainIfaceStatus.Name = name
			interfaceStatusByMac[ifaceSpec.MacAddress] = domainIfaceStatus
		}
	}
}
```

And I mention it here again, I would move the call to this method to be just before line 629 after the already handled interfaces were removed from the status list.

Reply:
+ @EdDev since `indexSpecIfaceNamesByMac` was his request
Also can you 2nd opinion please if it should be in the original place or before block 629
considering that we delete from this struct (see please https://github.com/kubevirt/kubevirt/pull/4335#discussion_r521852349), and it means that the logic updates it 2 times, instead 1

Reply:
I don't understand what 3 places you're talking about. But the "delete" for the statuses list is exactly the reason I want you to move the code.
Your "workaround" code should be *after* the regular handling.

Reply:
the 3 
init
delete
update

i understand, but i just ask 2nd opinion first please, since sometimes Edy explained that places of update should be minimized

Reply:
Oh, you ask Eddy for second opinion. Np, let's wait for the doctor:)

Reply:
Originally I saw it as a generic operation that adds the name of the network if it is missing.
Now it is specific to SRIOV, and no one else uses the indexing, making the argument to call a single function that handles SRIOV specifics strong.

The fact that the function will do two things is a bit ugly, but I think we can live with it (comparing to the mess the `updateVMIStatus` function is in).

I'm good with the proposal.

Reply:
I assume you are also fine with the place of it (line 629), this was the main question,
Thanks